edgelet/iotedge/src/check/checks/host_connect_dps_endpoint.rs

```diff
@@ -102,3 +128,37 @@ pub fn resolve_and_tls_handshake(
 
     Ok(())
 }
+
+pub fn resolve_and_tls_handshake_proxy(
+    tls_hostname: String,
+    port: Option<u16>,
+    proxy: String,
+) -> impl Future<Item = (), Error = Error> {
+    futures::future::ok(())
+        .and_then({
+            move |_| -> Result<_, Error> {
+                let proxy_uri = Uri::from_str(&proxy)
+                    .with_context(|_| format!("Could not make proxi uri from {}", proxy))?;
+                let proxy_obj = Proxy::new(Intercept::All, proxy_uri);
```

Comment:
Need to handle username and password in `proxy_uri` by extracting them and calling `proxy_obj.set_authorization`. See `hyperwrap::uri_to_proxy` for an example.